projects/assets-library/src/icons/icon-registry.service.ts

```diff
@@ -1,16 +1,23 @@
 import { Injectable } from '@angular/core';
 import { MatIconRegistry } from '@angular/material/icon';
 import { DomSanitizer } from '@angular/platform-browser';
-import { isEmpty } from 'lodash-es';
+import { isEmpty, orderBy } from 'lodash-es';
 import { Observable } from 'rxjs';
 
 @Injectable({ providedIn: 'root' })
 export class IconRegistryService {
   private static readonly SVG_ICON_PREFIX: string = 'svg:';
 
+  private registeredIcons: string[] = [];
+
   public constructor(private readonly matIconRegistry: MatIconRegistry, private readonly sanitizer: DomSanitizer) {}
 
   public registerSvgIcon(iconInfo: SvgIconRegistrationInfo): void {
+    if (!this.registeredIcons.includes(iconInfo.key)) {
+      this.registeredIcons.push(iconInfo.key);
+      this.registeredIcons = orderBy(this.registeredIcons, key => key, 'asc');
```

Comment:
do we need to store them in order? if the order is important only in the read path, we could do it there instead of ordering for every insert.

Reply:
RIght :), removed  as we do not need it here